app/controllers/users_controller.rb

```diff
@@ -0,0 +1,59 @@
+class UsersController < ApplicationController
+  before_action :retrieve_users, :authenticate_user!
+  before_action :authenticate_user!, only: %i[fight add_experience]
+  skip_before_action :verify_authenticity_token
```

Comment:
In case you are providing JSON API you can disable this on the global level, otherwise, this action is not secure.